automerge-backend/src/change.rs

```diff
@@ -634,23 +634,25 @@ fn group_doc_change_and_doc_ops(
         }
     }
 
-    'outer: for op in ops.iter() {
-        let max_seq = *actor_max.get(&op.actor).ok_or_else(|| {
-            decoding::Error::ChangeDecompressFailed("Doc Op.Actor Invalid".into())
-        })?;
-        for seq in 1..max_seq {
-            // this is safe - invalid seq would have thrown an error earlier
-            let idx: usize = *change_actors.get(&(op.actor, seq)).unwrap();
-            // this is safe since I build the array above ^^
-            let change = &mut changes[idx];
-            if op.ctr <= change.max_op {
-                change.ops.push(op.clone());
-                continue 'outer;
+    for op in ops {
+        // binary search for our change
+        let actor_change_index = changes_by_actor.entry(op.actor).or_default();
```

Comment:
This `actor_change_index` should probably be `actor_change_indices` here and above.

Reply:
I know that is true in our implementation but is it always required? I remember seeing another comment, maybe in columnar, that they may not always be in causal order.

Reply:
Ill double check with martin but both implementations follow the same logic on doc save and use this same binary search.

Reply:
Ok, just something to make sure gets documented then.